lib/rack/timeout/legacy.rb

```diff
@@ -10,23 +10,24 @@ module Rack::Timeout::ClassLevelProperties
   module ClassMethods
     attr_accessor :service_timeout, :wait_timeout, :wait_overtime, :service_past_wait
     alias_method :timeout=, :service_timeout=
+  end
 
+  module CallWithDeprecation
     [ :service_timeout=,
       :timeout=,
       :wait_timeout=,
       :wait_overtime=,
       :service_past_wait=,
     ].each do |isetter|
-      setter = instance_method(isetter)
-      define_method(isetter) do |x|
+      define_method(isetter) do |*|
         defined?(Rails) or warn "`Rack::Timeout.#{isetter}`: class-level settings are deprecated. See README for examples on using the middleware initializer instead."
-        setter.bind(self).call(x)
+        super
```

Comment:
As I wrote this to commit message, ruby says: rack-timeout/lib/rack/timeout/legacy.rb:21: warning: method redefined; discarding old wait_timeout=

This is because rack-timeout defines the same method right after the original one is defined.

I think this change won't break the behavior of the original method, but we can do in other ways.